### PR TITLE
fix(storefront): STRF-2456 Fixing Schema Org meta tags

### PR DESCRIPTION
**Pull request by davidwrpayne**
#### What?

We are adding schema.org offer tags on div's that schema.org doesn't care about what price was
it only cares about what price currently is afaict  I need to validate still that price-range schema works.

- [x] Test case about simple product
- [x] Test case about complex product
- [x] Test case about pricelist enabled product
- [x] Test case about Product with varying Tax inclusion settings.

#### Tickets / Documentation

- [STRF-2456](https://jira.bigcommerce.com/browse/STRF-2456)

#### Screenshots

Fixed product schema with "was" and "currently" prices
screen shot 2018-05-11 at 5 10 11 pm

### CODE REVIEW

**Comment by davidwrpayne** [comment]
@mattolson @Ubersmake @Souldjinn @mjschock This is ready for review

File: templates/components/products/price-range.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{#and price_range.min.with_tax price_range.max.with_tax}}
-    <div class="price-section price-section--withTax" {{#if schema_org}}itemprop="offers" itemscope itemtype="http://schema.org/Offer"{{/if}}>
```
[the remark this answers is not in the document]
Comment (davidwrpayne) [reply templates/components/products/price-range.html]: I believe the div with class can stay however the schema meta needs to be removed. Will fix.

[the remark this answers is not in the document]
Comment (mattolson) [reply templates/components/products/price-range.html]: I don't understand. This is embedded in `templates/components/products/price.html` without any "offers" wrapper so why is it ok to remove this? If you are removing it here, should you add the wrapper div below?

**Comment by davidwrpayne** [comment]
Much slimmer changeset.

File: templates/components/products/price-range.html [hunk 2]
```diff
@@ -2,8 +2,10 @@
     <div class="price-section price-section--withTax" {{#if schema_org}}itemprop="offers" itemscope itemtype="http://schema.org/Offer"{{/if}}>
         <span data-product-price-with-tax class="price">{{price_range.min.with_tax.formatted}} - {{price_range.max.with_tax.formatted}}</span>
         {{#and price_range.min.without_tax price_range.max.without_tax}}
-            <abbr title="{{lang 'products.excluding_tax'}}">{{lang 'products.price_with_tax' tax_label=price_range.min.tax_label}}</abbr>
-        {{else if schema_org}}
+            <abbr title="{{lang 'products.including_tax'}}">{{lang 'products.price_with_tax' tax_label=price_range.min.tax_label}}</abbr>
+
```
Comment (mattolson) [inline templates/components/products/price-range.html]: Extra line

File: templates/components/products/price-range.html [hunk 3]
```diff
@@ -2,8 +2,10 @@
...
+        {{/and}}
+            {{#if schema_org}}
```
Comment (mattolson) [inline templates/components/products/price-range.html]: Extra indent

File: templates/components/products/price-range.html [hunk 4]
```diff
@@ -13,7 +15,7 @@
                     <meta itemprop="priceCurrency" content="{{currency_selector.active_currency_code}}">
                     <meta itemprop="valueAddedTaxIncluded" content="true">
                 </div>
-        {{/and}}
+            {{/if}}
```
Comment (mattolson) [inline templates/components/products/price-range.html]: Extra indent

File: templates/components/products/product-view.html [hunk 5]
```diff
@@ -28,6 +29,7 @@ <h2 class="productView-brand"{{#if schema}} itemprop="brand" itemscope itemtype=
                 {{/or}}
             </div>
             {{{region name="product_below_price"}}}
+            {{#if product.rating}}
```
Comment (mattolson) [inline templates/components/products/product-view.html]: This seems unrelated to the change. Is there another bug you are trying to fix?

File: templates/components/products/product-view.html [hunk 6]
```diff
@@ -11,10 +11,11 @@
     <section class="productView-details">
         <div class="productView-product">
             <h1 class="productView-title" {{#if schema}}itemprop="name"{{/if}}>{{product.title}}</h1>
-
-            <h2 class="productView-brand"{{#if schema}} itemprop="brand" itemscope itemtype="http://schema.org/Brand"{{/if}}>
-                <a href="{{product.brand.url}}"{{#if schema}} itemprop="url"{{/if}}><span{{#if schema}} itemprop="name"{{/if}}>{{product.brand.name}}</span></a>
-            </h2>
+            {{#if product.brand}}
```
Comment (mattolson) [inline templates/components/products/product-view.html]: This seems unrelated to the change. Is there another bug you are trying to fix?

Comment (davidwrpayne) [reply templates/components/products/product-view.html]: yes. we were getting brand sections showing up on products without brands.

File: templates/components/products/product-view.html (hunk 5 above)
Comment (davidwrpayne) [reply templates/components/products/product-view.html]: yes we were getting rating sections showing up on products without ratings

Comment (mattolson) [reply templates/components/products/product-view.html]: Ok, with the introduction of this new block, please indent the contents
